# HDSC Fall 22 Graded Quiz - Python Programming - Quiz Results

QUESTION - SOLUTION

# To import the dataset from the url given and derive the mean and standard deviation per year:

In [86]:
import pandas as pd
url=("https://github.com/HamoyeHQ/HDSC-Introduction-to-Python-for-machine-learning/files/7768140/FoodBalanceSheets_E_Africa_NOFLAG.csv")
quiz_data=pd.read_csv(url, encoding='Latin-1')
quiz_data.describe(include='all')

Area Code   Area     Item Code                     Item  \
count   60943.000000  60943  60943.000000                    60943   
unique           NaN     49           NaN                      119   
top              NaN  Kenya           NaN  Milk - Excluding Butter   
freq             NaN   1560           NaN                     1262   
mean      134.265576    NaN   2687.176706                      NaN   
std        72.605709    NaN    146.055739                      NaN   
min         4.000000    NaN   2501.000000                      NaN   
25%        74.000000    NaN   2562.000000                      NaN   
50%       136.000000    NaN   2630.000000                      NaN   
75%       195.000000    NaN   2775.000000                      NaN   
max       276.000000    NaN   2961.000000                      NaN   

        Element Code                   Element         Unit          Y2014  \
count   60943.000000                     60943        60943   59354.000000   
unique           NaN                        18            5            NaN   
top              NaN  Domestic supply quantity  1000 tonnes            NaN   
freq             NaN                      5295        40933            NaN   
mean     3814.856456                       NaN          NaN     134.196282   
std      2212.007033                       NaN          NaN    1567.663696   
min       511.000000                       NaN          NaN   -1796.000000   
25%       684.000000                       NaN          NaN       0.000000   
50%      5142.000000                       NaN          NaN       0.090000   
75%      5511.000000                       NaN          NaN       8.340000   
max      5911.000000                       NaN          NaN  176405.000000   

                Y2015          Y2016          Y2017          Y2018  
count    59395.000000   59408.000000   59437.000000   59507.000000  
unique            NaN            NaN            NaN            NaN  
top               NaN            NaN            NaN            NaN  
freq              NaN            NaN            NaN            NaN  
mean       135.235966     136.555222     140.917765     143.758381  
std       1603.403984    1640.007194    1671.862359    1710.782658  
min      -3161.000000   -3225.000000   -1582.000000   -3396.000000  
25%          0.000000       0.000000       0.000000       0.000000  
50%          0.080000       0.080000       0.100000       0.070000  
75%          8.460000       8.430000       9.000000       9.000000  
max     181137.000000  185960.000000  190873.000000  195875.000000

The mean and standard deviation across the whole dataset for the year 2017 to 2 decimal places = 140.92 and 1671.86

# The total number of unique countries in the dataset is 49. 

In [87]:
quiz_data.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

# The total number of missing data in 2014 = 1589

In [96]:
#The percentage of missing data in 2014 to 3 decimal places
percent2014=((1589)/(59354))*100
print('The percentage of missing data in 2014 is')
percent2014

The percentage of missing data in 2014 is


2.6771573946153584

# The total sum of Wine produced in 2015 and 2018 

In [97]:
quiz_data.groupby('Item').sum('all')

Area Code  Item Code  Element Code      Y2014      Y2015  \
Item                                                                            
Alcohol, Non-Food        38447     752497       1515090    2403.00    2180.00   
Alcoholic Beverages      75016    1649136       2171381  102410.11   98783.72   
Animal Products          18060     397035         90990   11935.65   11811.73   
Animal fats              85279    1882494       2567303  209460.54  200675.72   
Apples and products      69280    1352989       1920418    9499.23   10559.15   
...                        ...        ...           ...        ...        ...   
Vegetables, Other        77779    1518715       2275409  155038.96  158104.08   
Vegetal Products         18060     391905         90990  107145.19  107064.17   
Wheat and products       87073    1639683       2654934  232670.13  234710.51   
Wine                     66206    1319535       1818328    4497.36    4251.81   
Yams                     50798     970905       1525745  200396.96  203151.78   

                         Y2016      Y2017      Y2018  
Item                                                  
Alcohol, Non-Food      2223.00    2348.00    2293.00  
Alcoholic Beverages   96958.75   95581.06   97847.27  
Animal Products       11661.69   11547.65   11578.61  
Animal fats          183314.22  269617.53  269648.27  
Apples and products    9853.26   10198.90    9640.51  
...                        ...        ...        ...  
Vegetables, Other    156218.90  157752.59  163987.21  
Vegetal Products     106792.72  107655.20  107775.39  
Wheat and products   239377.07  240047.62  242645.19  
Wine                   3872.09    4178.02    4039.32  
Yams                 220626.20  229174.59  221272.09  

[119 rows x 8 columns]

The total sum of Wine produced in 2015 and 2018 respectively is 4251.81 and 4038.32

# Quiz Code Solution 1

In [98]:
S = [['him', 'sell'], [90, 28, 43]]
S[0][1][1]


'e'

# Quiz Code Solution 2

In [101]:
lst = [[35, 'Portugal', 94], [33, 'Argentina', 93], [30 , 'Brazil', 92]]
col = ['Age','Nationality','Overall']
pd.DataFrame(lst, columns = col, index = [i for i in range(1,4)])

Age Nationality  Overall
1   35    Portugal       94
2   33   Argentina       93
3   30      Brazil       92

# Quiz Code Solution 3

In [104]:
y = [(2, 4), (7, 8), (1, 5, 9)]
x = y[1][1]
x

8

# Quiz Code Solution 4

In [105]:
my_tuppy = (1,2,5,8)

my_tuppy[2] = 6

TypeError: 'tuple' object does not support item assignment

In [106]:
food=quiz_data[["Y2017","Area"]]
food.head()

Y2017     Area
0  41389.00  Algeria
1      0.00  Algeria
2   3341.00  Algeria
3     92.82  Algeria
4     80.19  Algeria

In [112]:
fd=quiz_data.groupby('Y2017')['Area']
fd.head()

0         Algeria
1         Algeria
2         Algeria
3         Algeria
4         Algeria
           ...   
60825    Zimbabwe
60827    Zimbabwe
60841    Zimbabwe
60844    Zimbabwe
60847    Zimbabwe
Name: Area, Length: 9502, dtype: object